src/server/scripts/EasternKingdoms/zone_tirisfal_glades.cpp

```diff
@@ -38,134 +36,161 @@ EndContentData */
 
 enum Calvin
 {
-    SAY_COMPLETE        = 0,
-    SPELL_DRINK         = 2639,                             // possibly not correct spell (but iconId is correct)
-    QUEST_590           = 590
+    SAY_COMPLETE = 0,
+    SPELL_FOOD = 7737,
+    QUEST_590 = 590
+};
+
+enum event
+{
+    EVENT_EMOTE = 1,
+    EVENT_TALK = 2,
+    EVENT_DRINK_AND_QUEST_COMQUEST = 3,
+    EVENT_END = 4,
 };
 
 class npc_calvin_montague : public CreatureScript
 {
 public:
-    npc_calvin_montague() : CreatureScript("npc_calvin_montague") { }
+    npc_calvin_montague() : CreatureScript("npc_calvin_montague") {}
```

Comment:
if you are rewriting a script this much, just use the new registry

Reply:
This still valid. Use RegisterCreatureAI instead

Reply:
I've looked at other scripts and tried to use
 ```
void AddSC_tirisfal_glades()
{   
RegisterCreatureAI(npc_calvin_montague);
}
```
But
` bool OnQuestAccept(Player* player, Creature* creature, Quest const* quest) override `
is not used in other scripts
OnQuestAccept cannot be used in other classes

Reply:
I've looked at other scripts and tried to use
 ```
void AddSC_tirisfal_glades()
{   
RegisterCreatureAI(npc_calvin_montague);
}
```
But
` bool OnQuestAccept(Player* player, Creature* creature, Quest const* quest) override `
is not used in other scripts
OnQuestAccept cannot be used in other classes